### PR TITLE
Migrate remaining audit exceptions to homebrew/core

### PR DESCRIPTION
**Pull request by Rylan12**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----

This PR migrates the remaining audit exceptions to homebrew/core.

This is pretty straight forward, on this end, but there will be a few additional notes in the corresponding homebrew/core PR: https://github.com/Homebrew/homebrew-core/pull/65712

Note that it is expected that CI will fail until https://github.com/Homebrew/homebrew-core/pull/65712 has been merged into homebrew/core and linuxbrew/core.

### CODE REVIEW

File: Library/Homebrew/test/dev-cmd/audit_spec.rb
Comment (MikeMcQuaid) [inline Library/Homebrew/test/dev-cmd/audit_spec.rb:1014]: We're checking these formulae exist a different way now, right? I'm 99% sure we are but just checking 😀

File: Library/Homebrew/formula_auditor.rb
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_auditor.rb:221]: Would be good to port over this `TODO` somehow?

File: Library/Homebrew/formula_auditor.rb
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_auditor.rb:364]: I wonder (in a later PR) if this sort of thing should/could be ported over too.

**Review by MikeMcQuaid — APPROVED** [review]
Great work here @Rylan12. I've been impressed not just with your work on this but how you've handled the incremental rollout 👏🏻

File: Library/Homebrew/formula_auditor.rb
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_auditor.rb:159]: Code this get a `CODEOWNERS` added in Homebrew/brew to request review from me and the TSC? We need to ensure that we only add entries here that are definitely shipped on macOS (as we've had maintainers add to this list incorrectly in the past).

File: Library/Homebrew/test/dev-cmd/audit_spec.rb
Comment (Rylan12) [reply Library/Homebrew/test/dev-cmd/audit_spec.rb:1014]: Yep! `brew audit` now checks these on a per-tap basis rather than `brew tests`. It's part of [`tap_auditor.rb`](https://github.com/Homebrew/brew/blob/d7f03bc2dd61e3495f0b48b6c8371f567c215304/Library/Homebrew/tap_auditor.rb#L52-L75).

File: Library/Homebrew/formula_auditor.rb
Comment (Rylan12) [reply Library/Homebrew/formula_auditor.rb:221]: Was thinking about opening an issue in homebrew/core. Thoughts? We can't really have comments in JSON files and it probably doesn't belong in this repo if the exception list is gone

File: Library/Homebrew/formula_auditor.rb
Comment (Rylan12) [reply Library/Homebrew/formula_auditor.rb:364]: May as well add it here. I think the best way to do it will to add all versioned formulae/casks to the `versioned_keg_only_allowlist.json` file. The only downside that I can think of with that is this won't be as dynamic anymore. We'll need to remote old versioned formulae and add new ones as they're added/removed. The good news is that `brew audit` should complain in either case which will prompt modifying the lists. Does that make sense?

File: Library/Homebrew/formula_auditor.rb
Comment (Rylan12) [reply Library/Homebrew/formula_auditor.rb:159]: No problem! I'll handle it in https://github.com/Homebrew/homebrew-core/pull/65712

**Comment by Rylan12** [comment]
Thanks, @MikeMcQuaid! It means a lot and I appreciate the kind words!

It sounds like there are no major concerns, so I think I'll go ahead and merge the homebrew/core PR (after making the changes discussed above) to get the ball rolling as it will need to get through linuxbrew/core as well.

**Comment by MikeMcQuaid** [comment]
Another nice-to-have here eventually (while I remember) would be having some of these checks fail if they are no longer needed to encourage their removal (like RuboCop does with disables/todos). This would ensure that these lists get pruned when they aren't needed rather than growing with stale things.

This should not block this PR!

File: Library/Homebrew/formula_auditor.rb
Comment (Rylan12) [reply Library/Homebrew/formula_auditor.rb:364]: This `adoptopenjdk` reference can be removed, right? `adoptopenjdk` is a cask so it isn't checked by `formula_auditor` (plus it doesn't have different versions)

Comment (MikeMcQuaid) [reply Library/Homebrew/formula_auditor.rb:364]: yup!

File: Library/Homebrew/formula_auditor.rb
Comment (MikeMcQuaid) [reply Library/Homebrew/formula_auditor.rb:221]: Could maybe just be another CODEOWNERS for now?

**Comment by Rylan12** [comment]
> Another nice-to-have here eventually (while I remember) would be having some of these checks fail if they are no longer needed to encourage their removal (like RuboCop does with disables/todos). This would ensure that these lists get pruned when they aren't needed rather than growing with stale things.

Definitely agree. I'll have to think of how we want to do this. As one idea, we could add a `--prune-exceptions` option to audit which, if passed, could run _all_ checks and compare audit failures with the exceptions. I'll have to think more about the specifics so this will come later.

File: Library/Homebrew/formula_auditor.rb
Comment (Rylan12) [reply Library/Homebrew/formula_auditor.rb:221]: Maybe I'm misunderstanding what you mean because I don't really see how having this in `CODEOWNERS` would help. It would notify whoever is an "owner" _when_ the list is changed but won't _prompt_ anyone to try to remove instances.  It could, though, serve as an extra check to make sure new formulae aren't added. So maybe both? If so, should the owners be you and TSC as well?

Side note, `@Homebrew/tsc` includes you (and will always include the project leader), right? So no need to add both for `CODEOWNERS` changes (unless you'd rather be explicit).

File: Library/Homebrew/formula_auditor.rb
Comment (Rylan12) [reply Library/Homebrew/formula_auditor.rb:159]: Done

**Comment by MikeMcQuaid** [comment]
> Definitely agree. I'll have to think of how we want to do this. As one idea, we could add a `--prune-exceptions` option to audit which, if passed, could run _all_ checks and compare audit failures with the exceptions. I'll have to think more about the specifics so this will come later.

👍🏻 on deciding later. I'd say in general when the audits are checked for validity that they could be quickly scanned to ensure that they all fail and, if not, complain. I think this could be default behaviour because it's pretty much always going to be desirable behaviour.

**Comment by Rylan12** [comment]
Just waiting on https://github.com/Homebrew/linuxbrew-core/pull/21668

**Comment by jonchang** [comment]
Homebrew/linuxbrew-core#21670 is merged.

**Comment by Rylan12** [comment]
Thanks, @jonchang!

**Pull request merged**
